ci-operator/step-registry/openshift/microshift/infra/aws/ec2/openshift-microshift-infra-aws-ec2-commands.sh

```diff
@@ -71,329 +74,7 @@ function save_stack_events_to_shared()
-        FromPort: 30000
-        ToPort: 32767
-        CidrIp: 0.0.0.0/0
-      - IpProtocol: tcp
-        FromPort: 30000
-        ToPort: 32767
-        CidrIpv6: ::/0
-      - IpProtocol: udp
-        FromPort: 30000
-        ToPort: 32767
-        CidrIp: 0.0.0.0/0
-      - IpProtocol: udp
-        FromPort: 30000
-        ToPort: 32767
-        CidrIpv6: ::/0
-      VpcId: !Ref RHELVPC
-
-  rhelLaunchTemplate:
-    Type: AWS::EC2::LaunchTemplate
-    Properties:
-      LaunchTemplateName: ${stack_name}-launch-template
-      LaunchTemplateData:
-        BlockDeviceMappings:
-        - DeviceName: /dev/sda1
-          Ebs:
-            VolumeSize: !FindInMap [VolumeSize, !Ref EC2Type, PrimaryVolumeSize]
-            VolumeType: gp3
-            Throughput: !FindInMap [VolumeSize, !Ref EC2Type, Throughput]
-            Iops: !FindInMap [VolumeSize, !Ref EC2Type, Iops]
-        - !If
-          - AddSecondaryVolume
-          - DeviceName: /dev/sdc
-            Ebs:
-              VolumeSize: !FindInMap [VolumeSize, !Ref EC2Type, SecondaryVolumeSize]
-              VolumeType: gp3
-          - !Ref AWS::NoValue
-
-  RHELInstance:
-    Type: AWS::EC2::Instance
-    Properties:
-      ImageId: !Ref AmiId
-      LaunchTemplate:
-        LaunchTemplateName: ${stack_name}-launch-template
-        Version: !GetAtt rhelLaunchTemplate.LatestVersionNumber
-      IamInstanceProfile: !Ref RHELInstanceProfile
-      InstanceType: !Ref HostInstanceType
-      NetworkInterfaces:
-      - AssociatePublicIpAddress: "True"
-        DeviceIndex: "0"
-        GroupSet:
-        - !GetAtt RHELSecurityGroup.GroupId
-        SubnetId: !Ref RHELPublicSubnet
-      Tags:
-      - Key: Name
-        Value: !Join ["", [!Ref Machinename]]
-      PrivateDnsNameOptions:
-        EnableResourceNameDnsARecord: true
-        HostnameType: resource-name
-      UserData:
-        Fn::Base64: !Sub |
-          #!/bin/bash -xe
-          echo "====== Authorizing public key ======" | tee -a /tmp/init_output.txt
-          echo "\${PublicKeyString}" >> /home/ec2-user/.ssh/authorized_keys
-          # Use the same defaults as OCP to avoid failing requests to apiserver, such as
-          # requesting logs.
-          echo "====== Updating inotify =====" | tee -a /tmp/init_output.txt
-          echo "fs.inotify.max_user_watches = 65536" >> /etc/sysctl.conf
-          echo "fs.inotify.max_user_instances = 8192" >> /etc/sysctl.conf
-          sysctl --system |& tee -a /tmp/init_output.txt
-          sysctl -a |& tee -a /tmp/init_output.txt
-
-Outputs:
-  InstanceId:
-    Description: RHEL Host Instance ID
-    Value: !Ref RHELInstance
-  PrivateIp:
-    Description: The bastion host Private DNS, will be used for cluster install pulling release image
-    Value: !GetAtt RHELInstance.PrivateIp
-  PublicIp:
-    Description: The bastion host Public IP, will be used for registering minIO server DNS
-    Value: !GetAtt RHELInstance.PublicIp
-EOF
-
-
+cat "${cf_gen}" > "${cf_tpl_file}"
```

Comment:
I'm not sure we need another temporary file for download. Can we download directly to ${cf_tpl_file}?

Reply:
changed